entity-service/src/main/kotlin/com/egm/stellio/entity/service/EntityService.kt

```diff
@@ -131,12 +135,17 @@ class EntityService(
         val propertyValue = getPropertyValueFromMap(propertyValues, NGSILD_PROPERTY_VALUE)
             ?: throw BadRequestDataException("Key $NGSILD_PROPERTY_VALUE not found in $propertyValues")
 
+        val datasetId = getPropertyValueFromMapAsString(propertyValues, NGSILD_DATASET_ID_PROPERTY)
+        if (datasetId == null && neo4jRepository.hasPropertyWithDefaultInstance(EntitySubjectNode(entity.id), propertyKey))
+            throw BadRequestDataException("Property $propertyKey already has default instance")

```

Comment:
To optimize performances, we should think about a solution in which you check that the payload is valid (meaning only one default attribute, unique datasetIds) before launching the creation. What do you think @HoucemKacem @bobeal ?

Reply:
Also, it creates less noises in the already big EntityService.